homeassistant/components/homekit_controller/__init__.py

```diff
@@ -27,14 +27,23 @@ class HomeKitEntity(Entity):
 
     def __init__(self, accessory, devinfo):
         """Initialise a generic HomeKit device."""
-        self._available = True
         self._accessory = accessory
         self._aid = devinfo['aid']
         self._iid = devinfo['iid']
         self._features = 0
         self._chars = {}
         self.setup()
 
+        accessory.entities[(self._aid, self._iid)] = self
```

Comment:
We should avoid storing entities for central access in a shared container. It's more safe and decoupled to use our dispatch helper to signal the entity.

Reply:
This is now reverted to how it was before the PR - its now just a record of which (aid, iid) combinations we've already added HA entities for.